commit 61b1214b2401a31d8c23f126a8b610edd29ba8e1
Author: Frédéric Brégier <fredbregier@free.fr>
Date:   Thu Jul 25 15:19:42 2013 +0200

    Fix for first issue from #1652 on computation of time to wait in AbstractTrafficShapingHandler for Netty 4
    
    Fix for first issue from #1652 on computation of time to wait in AbstractTrafficShapingHandler for Netty 4, using the same formula than in Netty 3 (wrong place for parenthese).
    Was:
    
        (bytes * 1000 / limit - interval / 10) * 10;
    
    Becomes:
    
        (bytes * 1000 / limit - interval) / 10 * 10;

diff --git a/handler/src/main/java/io/netty/handler/traffic/AbstractTrafficShapingHandler.java b/handler/src/main/java/io/netty/handler/traffic/AbstractTrafficShapingHandler.java
index 1e03001..05586e0 100644
--- a/handler/src/main/java/io/netty/handler/traffic/AbstractTrafficShapingHandler.java
+++ b/handler/src/main/java/io/netty/handler/traffic/AbstractTrafficShapingHandler.java
@@ -205,7 +205,7 @@ public abstract class AbstractTrafficShapingHandler extends ChannelDuplexHandler
             // Time is too short, so just lets continue
             return 0;
         }
-        return (bytes * 1000 / limit - interval / 10) * 10;
+        return (bytes * 1000 / limit - interval) / 10 * 10;
     }
 
     @Override
